libsolidity/formal/SMTChecker.cpp

```diff
@@ -343,15 +348,14 @@ void SMTChecker::endVisit(FunctionCall const& _funCall)
 	{
 		solAssert(args.size() == 1, "");
 		solAssert(args[0]->annotation().type->category() == Type::Category::Bool, "");
-		checkCondition(!(expr(*args[0])), _funCall.location(), "Assertion violation");
-		m_interface->addAssertion(expr(*args[0]));
+		checkCondition(currentPathConditions() && !(expr(*args[0])), _funCall.location(), "Assertion violation");
```

Comment:
Shouldn't we still add this as a condition? In case the SMT solver cannot verify it, this should remove any subsequent errors (note that the assertion will be a runtime check).

Reply:
You're right. Added back.